react/OrderQueue.tsx

```diff
@@ -33,15 +33,15 @@ const OrderQueueContext = createContext<Context | undefined>(undefined)
 export const useQueueStatus = (listen: ListenFunction) => {
   const queueStatus = useRef(QueueStatus.FULFILLED)
 
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
Could you briefly explain the importance of this being `useLayoutEffect` instead of just `useEffect`?

Reply:
sure! I changed it mostly because the `queueStatus` ref wasn't in the "correct state" when the `useEffect`s in the `OrderFormProvider` component ran (if you added something in the queue in the first render it would still be with the `FULFILLED` status).

by changing from `useEffect` to `useLayoutEffect` here we effectively make this code run _before_ the other effects (since it synchronous) and have the correct status when the `useEffect`s in `OrderFormProvider` runs.